File: linear_search/linear_search.go
/* An implementation of Linear Search in Golang */

package main

import "fmt"

func linearSearch(array []int, num int) int {
	for i, item := range array {
		if item == num {
			return i
		}
	}
	return -1
}

func main() {

	fmt.Println("Linear search:")
	array := []int{0, 2, 4, 6, 8, 10, 12, 14, 16, 18}
	index := linearSearch(array, 10)
	if index == -1 {
		fmt.Println("Number not found")
	} else {
		fmt.Println("Index: ", index)
		fmt.Println("array[", index, "] = ", array[index])
	}
}

File: linear_search/ruby/linear_search.rb
def linear_search(array, key)
  i = 0
  while i < array.length
      if array[i] == key
        return "#{key} at index #{array.index(key)}"
      end
      i+=1
    end
    return -1
end

array = [1,6,7,3,4]
key = 3
puts linear_search(array, key)
